[cleanup] Remove insert_observers pass

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25217 [quant][graphmode] Integration tests for qconfig_dict
* #25127 [quant][graphmode] insert_quant_dequant work with qconfig_dict
* #25069 [graphmode] insert_observers use qconfig_dict
* #25000 [cleanup] Remove InsertQuantDeQuantNode
* **#24999 [cleanup] Remove insert_observers pass**
* #24998 [cleanup] Remove deprecated graph mode quantization tests
* #25209 [fix] skip tests if fbgemm is not supported for test_quantizer.py

Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001138](https://our.internmc.facebook.com/intern/diff/D17001138)

jerryzh168 committed:
[cleanup] Remove insert_observers pass

Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

ZolotukhinM commented on torch/csrc/jit/init.cpp:
```diff
@@ -137,20 +137,6 @@ void initJITBindings(PyObject* module) {
-            InsertObserverNodes(moduleObj, methodName, new_node);
-            // We don't need this node anymore, don't forget to remove it.
-            new_node->destroy();
-          })
```
Please follow up on the TODO below in this PR.

ZolotukhinM reviewed (approved):
👍

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001138](https://our.internmc.facebook.com/intern/diff/D17001138)

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001138](https://our.internmc.facebook.com/intern/diff/D17001138)

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001138](https://our.internmc.facebook.com/intern/diff/D17001138)

jerryzh168 committed:
Update on "[cleanup] Remove insert_observers pass"


Summary:
As described in previous PR, we are doing module level observer rather
than global observer now, so majority of code are deprecated. But we
still keeps some logic that is independent of this decision in the new
code.

Test Plan:
.

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001138](https://our.internmc.facebook.com/intern/diff/D17001138)
